### ISSUE TITLE
How to implement contract transfer

### ISSUE THREAD

**Issue by c0yax** [ethereum/go-ethereum#24514]
Like this
[bscscan.com](https://bscscan.com/tx/0xf6f24177727f3c0cb297bbe395520527942f41b33cd0c7873a627d236bb21252)

my code(I am currently only able to transfer money directly between wallets)
```
func Buy(client *ethclient.Client) {
	privateKey, err := crypto.HexToECDSA(h.privateKey)
	if err != nil {
		panic(err)
	}
	publicKey := privateKey.Public()
	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
	if !ok {
		panic(err)
	}
	// derivation from private key
	fromAddress := crypto.PubkeyToAddress(*publicKeyECDSA)
	// read random numbers for account transactions
	nonce, err := client.PendingNonceAt(context.Background(), fromAddress)
	if err != nil {
		panic(err)
	}
	value := big.NewInt(0)
	gasPrice, err := client.SuggestGasPrice(context.Background())
	if err != nil {
		panic(err)
	}
	toAddress := common.HexToAddress(h.toAddress)
	tokenAddress := common.HexToAddress(h.tokenAddress)
	transferFnSignature := []byte("transfer(address,uint256)")
	hash := sha3.NewLegacyKeccak256()
	hash.Write(transferFnSignature)
	methodID := hash.Sum(nil)[:4]
	// Left-fill the address of the sending token to 32 bytes
	paddedAddress := common.LeftPadBytes(toAddress.Bytes(), 32)

	// Int needs to be formatted as wei in big.
	amount := new(big.Int)
	amount.SetString(h.number, 10) // 1000 tokens
	// The amount of tokens also needs to be left filled to 32 bytes
	paddedAmount := common.LeftPadBytes(amount.Bytes(), 32)

	// add the filled address and the filled transfer amount to the byte slice
	var data []byte
	data = append(data, methodID...)
	data = append(data, paddedAddress...)
	data = append(data, paddedAmount...)

	// Build transaction transaction types
	tx := types.NewTransaction(nonce, tokenAddress, value, 300000, gasPrice, data)
	// Get the chain id from the client
	chainID, err := client.NetworkID(context.Background())
	if err != nil {
		panic(err)
	}
	// sign the transaction with the sender's private key. signTx method requires an EIP155 signer (EIP155 signer)
	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(chainID), privateKey)
	if err != nil {
		panic(err)
	}
	// broadcast Transactions
	err = client.SendTransaction(context.Background(), signedTx)
	if err != nil {
		panic(err)
	}
	fmt.Println(signedTx.Hash().Hex())
```

**Comment by karalabe** [comment]
You can use `abigen` to generate Go wrappers for contracts.

**Comment by maestroh1git** [comment]
Hi guys
Super new to Go and Geth and Ethereum
I deployed a contract to deposit and withdraw ether(with functions modified to onlyOwner) to ropsten with remix
I tested it with hardhat-ethers and its great, i created a new wallet instance with the provider and private key( for the deployer), then called the onlyOwner function and viola! transaction successful.

trying to replicate in go-ethers, after binding the newTransactor with the privatekey, it says "Execution Reverted"
binding the newTransactor with another privateKey gives an "Ownable: caller is not owner" error, so the function is good, buut why isnt the original deployer private key working to call the method as the Owner(deployer)?
